### PR TITLE
Activate turnstile on mainnet

### PR DESCRIPTION
**Pull request by bitcartel**
This PR enables [ZIP209](https://github.com/zcash/zips/blob/master/zip-0209.rst) support on mainnet, to mark blocks as invalid if they would lead to a turnstile violation in the Sprout or Sapling value pools.  

To test this PR, I performed the following manual tests:

1. Used RPC call `getblock` to verify the fallback Sprout value.
2. Individually changed the fallback Sprout block hash, block height and chain value, recompiling and relaunching the node, verifying that each individual change resulted in an error.  When the block hash and block height are incorrect, an error is logged to debug.log `FallbackSproutValuePoolBalance(): fallback block hash is incorrect`.  An incorrect chain value results in node termination with error: `void FallbackSproutValuePoolBalance(CBlockIndex*, const CChainParams&): Assertion '*pindex->nChainSproutValue == chainparams.SproutValuePoolCheckpointBalance()' failed.`
3. Ran the `Smoke Testing` described in PR #3885, on mainnet.
4. Launched zcashd with experimental feature `-developersetpoolsizezero` to manually trigger a turnstile violation in both Sprout and Sapling shielded pools.  The Sprout turnstile violation occurred after launch, due to chance, when the next incoming block 520786 contained a Sprout unshielding transaction. The Sapling turnstile violation was triggered after creating a Sapling unshielding transaction.

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -193,6 +193,14 @@ class CMainParams : public CChainParams {
                             //   total number of tx / (checkpoint block height / (24 * 24))
         };
 
+        // Hardcoded fallback value for the Sprout shielded value pool balance
+        // for nodes that have not reindexed since the introduction of monitoring
+        // in #2795.
+        nSproutValuePoolCheckpointHeight = 520633;
+        nSproutValuePoolCheckpointBalance = 22145062442933;
+        fZIP209Enabled = true;
+        hashSproutValuePoolCheckpointBlock = uint256S("0000000000c7b46b6bc04b4cbf87d8bb08722aebd51232619b214f7273f8460e");
```
Comment (LarryRuane) [inline src/chainparams.cpp]: Why do we not need similar for Sapling?

Comment (bitcartel) [reply src/chainparams.cpp]: By the time Sapling activated, the non-deprecated releases of Zcashd at that time were tracking and recording value deltas in the block delta.  For the lifetime of Sprout, this wasn't the case.  See point 1 here: https://github.com/zcash/zcash/pull/3885

**Comment by ebfull** [comment]
ACK!

I tested this a little differently than @bitcartel; I didn't use `-developersetpoolsizezero` but instead modified the wallet serialization logic to mimic an older node. As expected, the correct balance of the value pool was established near the end of the reindex. It was also consistent with the balance shown when the block index was reloaded.

**Comment by bitcartel** [comment]
@daira The checkpoint details can be verified with `zcash-cli getblock 520633`.

**Comment by bitcartel** [comment]
3 ACKs received: @zkbot r+

**Comment by ebfull** [comment]
(I also performed another `reindex` with no issues.)

**Pull request merged**